daemons/controld/controld_execd.c

```diff
@@ -574,17 +583,24 @@ append_restart_list(lrmd_event_data_t *op, struct ra_metadata_s *metadata,
         return;
     }
 
-    if (pcmk_is_set(metadata->ra_flags, ra_supports_reload)) {
-        /* Add any parameters with unique="1" to the "op-force-restart" list.
+    if (pcmk_all_flags_set(metadata->ra_flags,
```

Comment:
Does this really implement the logic from the heading?
Unless of course (!ra_supports_ocf_1_1 && ra_supports_reload_agent) is prevented by the meta-data-parser.

if (ra_supports_ocf_1_1) {
    if (ra_supports_reload_agent) {
        build_parameter_list( ... ra_param_reloadable ...)
    } else {
        // Resource does not support agent reloads
        return;
    }
} else {
    if (ra_supports_reload_agent) {
        // that one is questionable maybe
        build_parameter_list( ... ra_param_reloadable ...)
    } else if (ra_supports_reload) {
        build_parameter_list( ... ra_param_unique ...)
    } else {
        // Resource does not support reloads handled pre-1.1-compat
        return;
    }

Reply:
OCF 1.0 defined what the reload action should do, but Pacemaker's use of it was not compliant. OCF 1.1 defines both reload and reload-agent.

My intent here is that Pacemaker may remain out of compliance with OCF 1.0 for agents that support that, since it's existing practice, but we should be stricter about compliance with OCF 1.1 from the beginning, and only use actions as the standard intends (i.e., reload-agent and not reload for reloading parameters).

I was tempted to drop support for reload altogether for OCF 1.0 agents, since as far as I know, only the ocf:pacemaker dummy agents actually implemented the Pacemaker meaning. But I decided against surprising anyone using it in custom agents.

Reply:
That is more or less exactly what I wanted to bring up.
Doesn't the logic implemented enable the fallback mechanism exactly when sbdy has forgotten to bump the OCF version (as you say in your next reply) but in fact has implemented reload-agent and probably wants reload to work as initially described?
And yes I had tagged it as questionable already to support reload-agent in case it doesn't advertise OCF 1.1. And below it has a 'maybe' meaning that if not done here should be dropped below of course.
But - unless it isn't done already somewhere else where I don't remember - as we've identified advertising reload-agent without 1.1 as a possible bug in the agent that we don't want to obfuscate it might deserve a warning as to surface even quicker.

Reply:
I lean against warning here since it will be called for every action result.

But we could be stricter in the meta-data parsing, and only set reload/reload-agent support when the appropriate standard is supported. Then I could drop the ocf_1_1 flag altogether and just look at the action support.

Reply:
Yep - that is definitely the better place to issue that info instead of having to memorize that it has been issued already or flood the logs doing it every time here.
Btw. what about the `ra_supports_reload` flag? That is just for the pacemaker legacy handling and not for `reload` in the original OCF 1 sense and now OCF 1.1 sense - right?
Maybe easier to understand if we would call it `ra_supports_legacy_reload` or something.